sync/src/types/header_map/backend_sled.rs

```diff
@@ -0,0 +1,104 @@
+use super::KeyValueBackend;
+use crate::types::HeaderView;
+use ckb_types::{packed::Byte32, prelude::*};
+use sled::Db;
+use std::path;
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering;
+use tempfile::TempDir;
+
+pub(crate) struct SledBackend {
+    count: AtomicUsize,
```

Comment:
Can we remove `len` and `is_empty` from `KeyValueBackedn`?

Reply:
No, this is used for short-circuiting, when the count is 0 we can skip the db operation, this will be reflected in the profiling.
If you mean can we use `sled::len` or `sled::is_empty` , although I haven't actually tried it, but it shouldn't work either, performs a full O(n) scan under the hood.
```
   pub fn len(&self) -> usize {
        self.iter().count()
    }

    /// Returns `true` if the `Tree` contains no elements.
    pub fn is_empty(&self) -> bool {
        self.iter().next().is_none()
    }

   pub(crate) fn next_inner(&mut self) -> Option<<Self as Iterator>::Item> {
        let guard = pin();
        let (mut pid, mut node) = if let (true, Some((pid, node))) =
            (self.going_forward, self.cached_node.take())
        {
            (pid, node)
        } else {
            let view =
                iter_try!(self.tree.view_for_key(self.low_key(), &guard));
            (view.pid, view.deref().clone())
        };

        for _ in 0..MAX_LOOPS {
            if self.bounds_collapsed() {
                return None;
            }

            if !node.contains_upper_bound(&self.lo) {
                ...........
            } else if !node.contains_lower_bound(&self.lo, true) {
               ........
            } else {
                  ........
            }
        }
   
    }
```